src/main/java/Application/Utilities/OpenCVUtil.java

```diff
@@ -0,0 +1,138 @@
+package Application.Utilities;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.opencv.core.Mat;
+import org.opencv.core.CvType;
+
+import java.io.IOException;
+import java.io.InvalidObjectException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class OpenCVUtil {
+    public static String serializeMat(Mat subject) throws InvalidObjectException, JsonProcessingException {
+        int type = subject.type();
+        int rows = subject.rows();
+        int cols = subject.cols();
+        int total = (int)subject.total();
+        int elemSize = (int)subject.elemSize();
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode result = mapper.createObjectNode();
+        ArrayNode dataNode = mapper.createArrayNode();
+        String returnedResult = "";
+
+        result.put("type", type);
+        result.put("rows", rows);
+        result.put("cols", cols);
+
+        if(type == CvType.CV_32S || type == CvType.CV_32SC2 || type == CvType.CV_32SC3 || type == CvType.CV_16S) {
+            int[] data = new int[cols * rows * elemSize];
+            subject.get(0, 0, data);
```

Comment:
it is from OpenCV Java, I'm not really sure how it works since the documentation doesn't say anything except for the parameter, but it seems the get is used to set the data field for Mat

multiple data types were used because each OpenCV type uses specific data types so I added everything to accomodate if we want to add new functionalities in the future